### ISSUE TITLE
java.lang.NullPointerException: null cannot be cast to non-null type kotlin.Boolean

### ISSUE THREAD

**Issue by arpitnnd** [razorpay/razorpay-android-sample-app#319]
## Description
Getting an NPE a few seconds after every checkout open().

compileSdk 33
targetSdk 33
minSdk 23
Kotlin 1.7.0
androidx.appcompat:appcompat:1.4.2
com.google.android.material:material:1.6.1

## Razorpay SDK Version :
com.razorpay:checkout:1.6.22

## Android OS Version :
Android 12L

### What you did:
Set up a payment & initiate it.

### What happened:
```
2022-07-07 17:26:50.288 25411-25411 System.err              com.foo                  W  java.lang.ClassNotFoundException: com.foo.BuildConfig
2022-07-07 17:26:50.289 25411-25411 System.err              com.foo                  W  	at java.lang.Class.classForName(Native Method)
2022-07-07 17:26:50.289 25411-25411 System.err              com.foo                  W  	at java.lang.Class.forName(Class.java:454)
2022-07-07 17:26:50.289 25411-25411 System.err              com.foo                  W  	at java.lang.Class.forName(Class.java:379)
2022-07-07 17:26:50.289 25411-25411 System.err              com.foo                  W  	at com.razorpay.OpinionatedSoln.getBuildConfigValue(OpinionatedSoln.kt:194)
2022-07-07 17:26:50.289 25411-25411 System.err              com.foo                  W  	at com.razorpay.OpinionatedSoln.showDialog(OpinionatedSoln.kt:98)
2022-07-07 17:26:50.289 25411-25411 System.err              com.foo                  W  	at com.razorpay.OpinionatedSoln.checkEnvVariablesForProject(OpinionatedSoln.kt:50)
2022-07-07 17:26:50.289 25411-25411 System.err              com.foo                  W  	at com.razorpay.OpinionatedSoln.checkIfVersionUpdateExists$lambda-0(OpinionatedSoln.kt:90)
2022-07-07 17:26:50.289 25411-25411 System.err              com.foo                  W  	at com.razorpay.OpinionatedSoln.$r8$lambda$2u-MILbRY-RVpUqiCdvHUhh9kls(Unknown Source:0)
2022-07-07 17:26:50.289 25411-25411 System.err              com.foo                  W  	at com.razorpay.OpinionatedSoln$$ExternalSyntheticLambda3.run(Unknown Source:4)
2022-07-07 17:26:50.289 25411-25411 System.err              com.foo                  W  	at com.razorpay.i_$z_.onPostExecute(Owl.java:5137)
2022-07-07 17:26:50.289 25411-25411 System.err              com.foo                  W  	at android.os.AsyncTask.finish(AsyncTask.java:771)
2022-07-07 17:26:50.289 25411-25411 System.err              com.foo                  W  	at android.os.AsyncTask.access$900(AsyncTask.java:199)
2022-07-07 17:26:50.289 25411-25411 System.err              com.foo                  W  	at android.os.AsyncTask$InternalHandler.handleMessage(AsyncTask.java:788)
2022-07-07 17:26:50.289 25411-25411 System.err              com.foo                  W  	at android.os.Handler.dispatchMessage(Handler.java:106)
2022-07-07 17:26:50.289 25411-25411 System.err              com.foo                  W  	at android.os.Looper.loopOnce(Looper.java:201)
2022-07-07 17:26:50.289 25411-25411 System.err              com.foo                  W  	at android.os.Looper.loop(Looper.java:288)
2022-07-07 17:26:50.289 25411-25411 System.err              com.foo                  W  	at android.app.ActivityThread.main(ActivityThread.java:7877)
2022-07-07 17:26:50.289 25411-25411 System.err              com.foo                  W  	at java.lang.reflect.Method.invoke(Native Method)
2022-07-07 17:26:50.289 25411-25411 System.err              com.foo                  W  	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:550)
2022-07-07 17:26:50.289 25411-25411 System.err              com.foo                  W  	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
2022-07-07 17:26:50.289 25411-25411 System.err              com.foo                  W  Caused by: java.lang.ClassNotFoundException: Didn't find class "com.foo.BuildConfig" on path: DexPathList[[dex file "/data/data/com.foo/code_cache/.overlay/base.apk/classes13.dex", zip file "/data/app/~~1D2HU5lI0Ycn-ZYmJuUijA==/com.foo-ABlD76E8zHKb44M4yrXUPA==/base.apk"],nativeLibraryDirectories=[/data/app/~~1D2HU5lI0Ycn-ZYmJuUijA==/com.foo-ABlD76E8zHKb44M4yrXUPA==/lib/arm64, /system/lib64, /system_ext/lib64, /product/lib64]]
2022-07-07 17:26:50.289 25411-25411 System.err              com.foo                  W  	at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:218)
2022-07-07 17:26:50.289 25411-25411 System.err              com.foo                  W  	at java.lang.ClassLoader.loadClass(ClassLoader.java:379)
2022-07-07 17:26:50.289 25411-25411 System.err              com.foo                  W  	at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
2022-07-07 17:26:50.289 25411-25411 System.err              com.foo                  W  	... 20 more
2022-07-07 17:26:50.289 25411-25411 AndroidRuntime          com.foo                  D  Shutting down VM
2022-07-07 17:26:50.294 25411-25411 AndroidRuntime          com.foo                  E  FATAL EXCEPTION: main
                                                                                                    Process: com.foo, PID: 25411
                                                                                                    java.lang.NullPointerException: null cannot be cast to non-null type kotlin.Boolean
                                                                                                    	at com.razorpay.OpinionatedSoln.showDialog(OpinionatedSoln.kt:99)
                                                                                                    	at com.razorpay.OpinionatedSoln.checkEnvVariablesForProject(OpinionatedSoln.kt:50)
                                                                                                    	at com.razorpay.OpinionatedSoln.checkIfVersionUpdateExists$lambda-0(OpinionatedSoln.kt:90)
                                                                                                    	at com.razorpay.OpinionatedSoln.$r8$lambda$2u-MILbRY-RVpUqiCdvHUhh9kls(Unknown Source:0)
                                                                                                    	at com.razorpay.OpinionatedSoln$$ExternalSyntheticLambda3.run(Unknown Source:4)
                                                                                                    	at com.razorpay.i_$z_.onPostExecute(Owl.java:5137)
                                                                                                    	at android.os.AsyncTask.finish(AsyncTask.java:771)
                                                                                                    	at android.os.AsyncTask.access$900(AsyncTask.java:199)
                                                                                                    	at android.os.AsyncTask$InternalHandler.handleMessage(AsyncTask.java:788)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:106)
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:201)
                                                                                                    	at android.os.Looper.loop(Looper.java:288)
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:7877)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:550)
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
```

## Steps To Reproduce
1. Create an order on the server
2. Initiate a checkout payment on Android via SDK

## Code example, screenshot, or link to a repository:
```
Checkout checkout = new Checkout();
checkout.setKeyID(BuildConfig.RAZORPAY_KEY);
checkout.setImage(R.drawable.logo);
PayloadHelper payloadHelper = new PayloadHelper(
        "INR",
        (int) (amountPayable * 100),
        razorPayOrderId
);
payloadHelper.setName("Foo");
payloadHelper.setDescription("Order #" + orderId);
payloadHelper.setColor(context.getString(R.color.primaryColor).replace("#ff", "#"));
JSONObject notes = new JSONObject();
try {
    notes.put("merchant_order_id", orderId);
    payloadHelper.setNotes(notes);
} catch (Exception e) {
    Logger.e("Error in starting Razorpay Checkout:\n", e);
}
JSONObject options = payloadHelper.getJson();
checkout.open(activity, options);
```
```
Checkout checkout = new Checkout();
checkout.setKeyID(BuildConfig.RAZORPAY_KEY);
checkout.setImage(R.drawable.logo);
try {
    JSONObject options = new JSONObject();
    JSONObject notes = new JSONObject();
    options.put("name", "Foo");
    options.put("description", "Order #" + orderId); //intent
    options.put("currency", "INR");
    options.put("amount", amountPayable * 100);
    options.put("order_id", razorPayOrderId);
    options.put("theme.color",
            context.getString(R.color.primaryColor).replace("#ff", "#"));
    notes.put("merchant_order_id", orderId);
    options.put("notes", notes);
    checkout.open(activity, options);
} catch (Exception e) {
    Logger.e("Error in starting Razorpay Checkout:\n", e);
}
```

**Comment by chiragpatel92** [comment]
I am facing the same issue. any update here?

**Comment by tusharquantsapp** [comment]
I am facing the same issue. even my live application is Crashing .

**Comment by vivekshindhe** [comment]
Fixed in v1.6.23. Thank you for your patience.
